reVX/offshore/dist_to_ports.py

```diff
@@ -397,21 +431,227 @@ def run(cls, ports, excl_fpath, input_dist_layer='dist_to_coast',
         logger.info('Computing least cost distance to ports in {}'
                     .format(ports))
         dtp = cls(ports, excl_fpath, input_dist_layer=input_dist_layer)
-        if output_dist_layer is None:
-            output_dist_layer = os.path.basename(ports).split('.')[0]
 
         dist_to_ports = None
         if update_layer:
             hsds = check_res_file(excl_fpath)[1]
             with ExclusionLayers(excl_fpath, hsds=hsds) as tif:
-                if output_dist_layer in tif.layers:
+                if output_dist_layer in tif:
                     dist_to_ports = tif[output_dist_layer]
 
         if dist_to_ports is not None:
             logger.info('Updating exising layer {}'.format(output_dist_layer))
 
         dist_to_ports = dtp.least_cost_distance(dist_to_ports=dist_to_ports,
                                                 max_workers=max_workers)
-        dtp.save_as_layer(output_dist_layer, dist_to_ports, chunks=chunks)
+
+        if output_dist_layer:
+            dtp.save_as_layer(output_dist_layer, dist_to_ports, chunks=chunks)
 
         return dist_to_ports
+
+
+class AssemblyAreas:
```

Comment:
I think this class and its CLI should be in their own files. That would make it much easier to see the available utilities at a glance. otherwise you need to run the cli, scroll through this file, or know exactly what you're looking for.